### ISSUE TITLE
Allowing strides with dilated convolutions in tf.keras.layers.Conv2D

### ISSUE THREAD

**Issue by EmptySamurai** [tensorflow/tensorflow#48395]
**System information**
- TensorFlow version (you are using): v2.4.1
- Are you willing to contribute it (Yes/No): Depends on acceptable solution

**Describe the feature and the current behavior/state.**
At this point in time `tf.keras.layers.Conv2D` doesn't support dilated convolutions with strides > 1, which would be a nice feature to have. This limitation seems to be a limitation of Keras Conv2D, not a Tensorflow. `tf.nn.conv2d` supports dilated convolutions with strides, but all Keras Conv layers use generic `tf.nn.convolution`, which doesn't support dilated convolutions with strides.

The most natural solution would probably be creating something like `_conv_implementation` method in base Conv layer and override it using `tf.nn.conv1d`, `tf.nn.conv2d`, `tf.nn.conv3d` in corresponding Keras layers. 

**Will this change the current api? How?**
It seems that `tf.nn.convolution` and `tf.nn.conv1d`, `tf.nn.conv2d`, `tf.nn.conv3d` behave identically, so it shouldn't change API or break back-compatibility (except now dilated convolutions will be able to use strides).

**Who will benefit with this feature?**
Anybody who wants to use convolutional neural networks with large receptive field but little computational overhead.

**Comment by EmptySamurai** [comment]
I found that check here 
https://github.com/tensorflow/tensorflow/blob/9a2545aecc498e7a5e8bf5d63c47af59e1550111/tensorflow/python/ops/nn_ops.py#L1135
seems redundant as nn.conv2d supports gradient for dilations for non TPU devices
